lib/msg-format.c

```diff
@@ -72,6 +72,8 @@ msg_format_postprocess_message(MsgFormatOptions *options, LogMessage *msg)
           p++;
         }
     }
+  if (options->flags & LP_LOCAL)
+    msg->flags |= LF_LOCAL;
```

Comment:
I would move both `local` and `assume-utf8` flags to preprocess.
Reasons:
- syslog format handler in legacy-hander uses the `LF_LOCAL` flag
- regardles of `no-parse` flag, the post-processing is done unconditionally, so it doesn't matter if it's in pre- or post-processing.

Reply:
I think this is the reason that kira fails, as one test with flag `kernel` fails.

Reply:
Can you elaborate that?

Reply:
hmmm. I have now pushed a WIP patch on top, which clears all the flags in log_msg_clear(). I still have to give some more thoughts on the behavior, but that means that the error message becomes completely uncoupled with the original one.